bindings/python/pymongoarrow/builder.pyx

```diff
@@ -0,0 +1,94 @@
+# Copyright 2021-present MongoDB, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# Cython compiler directives
+# distutils: language=c++
+# cython: language_level=3
+from pyarrow.lib cimport *
+
+import numpy as np
+
+
+cdef class Int32Builder(_Weakrefable):
```

Comment:
`_Weakrefable` is this private? Is there a public class we can use instead?

Reply:
`_Weakrefable` is c-defined in `pyarrow.lib`. It just adds the ability for users to weakref this class. Strictly speaking, we don't need it at the moment in this implementation but `payarrow`'s classes are all Weakrefable so I included it here for consistency.

Reply:
The problem is that we're relying on a private api so let's not use _Weakrefable at all.:
```
cdef class _Weakrefable:
    cdef object __weakref__
```

Adding the `__weakref__` attribute allows these Python objects to be used with `weakref.ref()`. Do we expect to weakref() these classes? If not, then we shouldn't bother. If it turns out we need to support this then we can define _Weakrefable ourselves.

Reply:
Hmm ok I see your point. Removed.